.grit/patterns/re_match_walrus.md

```diff
@@ -0,0 +1,152 @@
+---
+title: Walrus operator for match in if
+---
+
+Use the walrus operator for snippets with the pattern:
+
+```python
+# Convert this:
+math = re.match(regex):
+if match:
+   ...
+
+# To this:
+if math := re.match(regex):
+   ...
+```
+
+Limitations:
+* If the match function is imported with an alias (e.g. `from re import match as m`), it will not be transformed.
+* When `re.match` is used, we do not check that `re` comes from `import re`.
+
+
+```grit
+engine marzano(0.1)
+language python
+
+pattern re_match_function() {
+    or {
+        `search`,
+        `match`,
+        `fullmatch`,
+    }
+}
+
+pattern imported_match_function() {
+    $func where {
+        $func <: re_match_function(),
+        $func <: imported_from(source = `re`),
+    },
+}
+
+// TODO: we should also check if `re` is imported
+pattern explicit_match_function() {
+    `re.$func` where {
+        $func <: re_match_function()
+    },
+    //$_ where {
+    //    $program <: contains `import re`,
+    //}
+}
+
+pattern match_function() {
+    or {
+        imported_match_function(),
+        explicit_match_function(),
+    }
+}
+
+if_statement($alternative, $condition, $consequence) as $if where {
+    $if <: after `$var = $match_func($regex)` => . where {
+        $match_func <: match_function()
+    },
+    $condition <: `$var` => `$var := $match_func($regex)`,
+    if ($alternative <: "") {
+        $block = $consequence,
+    }
+    else {
+        $separator = `\n`,
+        $block = join(list = [$consequence, $alternative], $separator),
+    }
+} => `if $condition:
+    $block`
+
+```
+
+## Transforms a log statement
```

Comment:
It's often easier to look at individual before/after pairs. I'd consider spitting this into multiple samples.

Reply:
Done, I've split all test cases.